straxen/contexts.py

```diff
@@ -261,56 +281,88 @@ def xenonnt_simulation(output_folder='./strax_data',
                     **straxen.contexts.xnt_common_config,),
         **straxen.contexts.xnt_common_opts, **kwargs)
     st.register(wfsim.RawRecordsFromFaxNT)
+    st.apply_cmt_version(f'global_{cmt_version}')
 
     if _forbid_creation_of is not None:
         st.context_config['forbid_creation_of'] += strax.to_str_tuple(_forbid_creation_of)
 
-    st.apply_cmt_version(f'global_{cmt_version}')
-    if not cmt_run_id:
-        raise ValueError('You have to specify a run_id which should be used to initialize '
-                         'the corrections.')
-    if _cmt_run_id_proc_only is None:
-        _cmt_run_id_proc_only = cmt_run_id
+    # doing sanity checks for cmt run ids for simulation and processing
+    if not (cmt_run_id_sim is None) and not (cmt_run_id_proc is None):
+        if not (cmt_run_id_sim==cmt_run_id_proc):
+            print("INFO : divergent CMT runs for simulation and processing")
+            print("    cmt_run_id_sim".ljust(25), cmt_run_id_sim)
+            print("    cmt_run_id_proc".ljust(25), cmt_run_id_proc)
+    elif (cmt_run_id_sim is None) and not (cmt_run_id_proc is None):
+        cmt_run_id_sim=cmt_run_id_proc
+    elif not (cmt_run_id_sim is None) and (cmt_run_id_proc is None):
+        cmt_run_id_proc=cmt_run_id_sim
+    else:
+        raise RuntimeError("Trying to set both cmt_run_id_sim and cmt_run_id_proc to None")
```

Comment:
```suggestion
    # doing sanity checks for cmt run ids for simulation and processing
    if not (cmt_run_id_sim  and cmt_run_id_proc ):
        raise RuntimeError("Trying to set both cmt_run_id_sim and cmt_run_id_proc to None")
    if (cmt_run_id_sim and cmt_run_id_proc ) and (cmt_run_id_sim!=cmt_run_id_proc):
        warnings.warn(("INFO : divergent CMT runs for simulation and processing", 
                               "    cmt_run_id_sim".ljust(25), cmt_run_id_sim,
                               "    cmt_run_id_proc".ljust(25), cmt_run_id_proc))
    else:
        cmt_id = cmt_run_id_sim  or cmt_run_id_proc 
        cmt_run_id_sim  = cmt_id 
        cmt_run_id_proc = cmt_id  
```

This change you should test after committing.

Reply:
I more or less change it in that direction, but i kept it as print and not warning, this is not intended to be the warning. But this opens an important point - do we have a proper logger with configurable verbosity level?

Reply:
Yes strax and straxen have it both.